### ISSUE TITLE
Retire Data Element and Data Element CodeSet

### ISSUE THREAD

**Issue by AngelaFaulding** [NHSDigital/DataDictionaryPublication#554]
In the Test Branch, I am retiring the Data Element and Data Element CodeSet PATIENT DIAGNOSIS INDICATOR (DEMENTIA).
I have a couple of questions relating to this:

The Data Element is linked to the Attribute PATIENT DIAGNOSIS INDICATOR.

Should there be an Attribute in the Attributes Linked field in the profile box that I should remove. In this example there isn't. Is this because of #523?

![image](https://github.com/user-attachments/assets/6e021066-7966-4436-9cb4-cd181f08b046)

When I retire the Data Element, this is what is shown in the Orchestrator:

![image](https://github.com/user-attachments/assets/9ccf5281-7a5d-43f1-9630-8b18cd43f48c)

When I retired the CodeSet and refreshed the page, the National Codes remain, which is not correct.

Do we have to remove the Attribute Terminology to Data Element CodeSet links when we retire items? as per #552 . If so, how do we do this?

**Comment by AngelaFaulding** [comment]
Peter will fix this.

**Comment by pjmonks** [comment]
This issue has been fixed and deployed to TEST, TRAINING and LIVE. Please move this to the "Ready for NHS E Testing" state.

**Comment by pjmonks** [comment]
The fix was that retired Data Elements no longer show National/Default codes.

**Comment by AngelaFaulding** [comment]
@pjmonks - can you please answer this question that I asked above?

In the Test Branch (CR8008), I am retiring the Data Element and Data Element CodeSet PATIENT DIAGNOSIS INDICATOR (DEMENTIA).

The Data Element is linked to the Attribute PATIENT DIAGNOSIS INDICATOR.

Should there be an Attribute in the Attributes Linked field in the profile box that I should remove. In this example there isn't. Is this because of #523 ?

<img width="396" alt="image" src="https://github.com/user-attachments/assets/7d6ea551-9130-429a-ad03-144408b1258d">

Also, the Format/Length is still showing on the retired Data Element. Do we have to remove that from the profile?

<img width="429" alt="image" src="https://github.com/user-attachments/assets/91a94906-6b80-4d1c-b3e1-b244526e539f">

**Comment by pjmonks** [comment]
The "Linked Attributes" field in the profile can be ignored, Data Elements and Attributes are not linked this way.

If you are saying that the "Format/Length" section should not be shown in a retired Data Element then I will remove it. Is this the same for preparatory Data Elements too?

- MauroDataMapper-NHSD/mdm-plugin-nhs-data-dictionary#126

**Comment by AngelaFaulding** [comment]
@pjmonks - please talk to @jamesrwelch about this. The "Linked Attributes" field does link Data Elements and Attributes, see #523.
Only the retired text should display on Data Elements. This is the same for preparatory items.

**Comment by pjmonks** [comment]
I have deployed a fix to TEST to hide the "Format/Length" section on retired Data Elements.

**Comment by pjmonks** [comment]
Apologies, understanding what should and should not be visible for retired/preparatory items is not always clear to me. I will update the Element pages and change paper ASAP:

- MauroDataMapper-NHSD/mdm-plugin-nhs-data-dictionary#133

**Comment by KatePalmerLilley** [comment]
To be discussed on 30/09/2024

**Comment by AngelaFaulding** [comment]
30/09/24 - Peter is now happy to progress this.

**Comment by pjmonks** [comment]
It was agreed in the meeting on 30/09/2024 that retired item pages will contain:

* Short description
* Retired text template (instead of the item description)
* Change log (New)

They will _not_ include:

* Original description
* Format/Length (for Elements)
* National Codes (for Elements/Attributes)
* Default Codes (for Elements)
* Attributes (for Classes)
* Relationships (for Classes)
* Linked Attributes (for Elements)
* Linked Elements (for Attributes)
* Where Used
* Aliases
* Specifications (for Data Sets)

If an item is retired, it may _not_ appear as a link on another page.

**Comment by KatePalmerLilley** [comment]
@pjmonks in CR8007 I've retired attribute  ABDOMINAL X-RAY PERFORMED REASON:
<img width="479" alt="image" src="https://github.com/user-attachments/assets/f1cda779-51ab-476a-9ddb-1945319ecf9a">

However the link is still appearing on the Data Element:
<img width="322" alt="image" src="https://github.com/user-attachments/assets/41618154-f72c-4fac-a22c-9cfbf7958a8e">

**Comment by pjmonks** [comment]
@KatePalmerLilley The problem isn't that the retired item is showing in that list, it's that the entire list of linked attributes is wrong. This all traces back to #506, the Mauro semantic links between items does not seem to be working correctly when creating new branches.

Here are the links that Element preview page is showing:

![image](https://github.com/user-attachments/assets/3d87dce4-0e3f-426b-b195-e6d47f3e4eea)

Then the problem is that the first link points to this Attribute:

![image](https://github.com/user-attachments/assets/20c2c28b-ae6e-4e81-ac9b-05a0bc82adda)

And the second link points to this Element (which is incorrectly labelled in the preview as an Attribute):

![image](https://github.com/user-attachments/assets/5840962a-df92-4bcb-899c-7d981a82cff2)

Both of those links are pointing to the wrong branch, which is why retiring an attribute on your branch is having no affect.

I'm talking to @jamesrwelch about this, because I'm also confused as to how these links between items are meant to be working.

**Comment by KatePalmerLilley** [comment]
Kate to speak to Peter about retired Data Element BARCELONA CLINIC LIVER CANCER STAGE retired but appearing in a Cancer Outcomes and Services Data Set - Liver:

<img width="575" alt="image" src="https://github.com/user-attachments/assets/b5230dbb-4994-48bc-8781-eee2ea3ee703">

<img width="472" alt="image" src="https://github.com/user-attachments/assets/56d70252-e223-420e-b9b8-a550020dd601">

Following the link to the data element from the data set the item isn't retired:

<img width="542" alt="image" src="https://github.com/user-attachments/assets/6da4e869-a582-4b06-b7ad-69135f7c25ec">

**Comment by KatePalmerLilley** [comment]
07/10/2024 discussed with @pjmonks, he will fix the the data element appearing on the the data set. Kate to continue to test.

**Comment by pjmonks** [comment]
Agreed with @KatePalmerLilley, I will look at fixing this. See:

- MauroDataMapper-NHSD/mdm-plugin-nhs-data-dictionary#147

**Comment by AngelaFaulding** [comment]
14/10/24 - Peter has some work to do on this.

**Comment by pjmonks** [comment]
It was agreed that:

- If a data element was retired, then it may still appear in a data set so long as the label to that data also says "(Retired)"
- Removing a retired data element from a data set is a manual process managed by the Data Dictionary team.

**Comment by AngelaFaulding** [comment]
21/10/24 - Peter is still working on this.

**Comment by AngelaFaulding** [comment]
@pjmonks - when re retire a Data Element and Code Set, do we have to remove the links:

![image](https://github.com/user-attachments/assets/08e26a67-9da5-4cd8-9444-bcfea61ff696)

**Comment by pjmonks** [comment]
@KatePalmerLilley - I have deployed to TEST a new change which has links to data elements in data set specifications show they are retired correctly. You gave the example earlier of the "Cancer Outcomes and Services Data Set - Liver" data set under CR8007. This is what the specification looks like now:

![Image](https://github.com/user-attachments/assets/7b769c17-6014-4da9-832e-1460700d3d6c)

Clicking on the retired link takes me to the retired page preview:

![Image](https://github.com/user-attachments/assets/9fa1fdf6-0412-40d0-9790-acff51553269)

The retired link also looks correct in the publication of change papers and the final website.

**Comment by pjmonks** [comment]
> [@pjmonks](https://github.com/pjmonks) - when retiring a Data Element and Code Set, do we have to remove the links:
> 
> ![image](https://github.com/user-attachments/assets/08e26a67-9da5-4cd8-9444-bcfea61ff696)

No, you don't. If PATIENT DIAGNOSIS INDICATOR (DEMENTIA) was used in a data set, the hyperlink to it will now appear as "retired" and will take you to the retired page. The retired page only shows the description for standard retired items, nothing more - the Code Set link can remain because it will not be shown anymore.

**Comment by AngelaFaulding** [comment]
I need to check the Wiki for this.

**Comment by AngelaFaulding** [comment]
The Wiki says:
"Mauro does not expect links and relationships to be deleted.
Therefore, once a Data Element is retired, any Attributes linked to the Data Element should not be removed.
The Orchestrator, preview and live dictionary will not display the links to retired items."
The page is ready for testing by Kate.
